lib/dynflow/action/with_sub_plans.rb

```diff
@@ -170,10 +170,10 @@ def sub_plans_count(filter = {})
     end
 
     def notify_on_finish(plans)
-      suspend do |suspended_action|
+      suspend do
         plans.each do |plan|
           plan.finished.on_resolution! do |success, value|
-            suspended_action << SubPlanFinished[plan.id, success && (value.result == :success)]
+            plan_event(SubPlanFinished[plan.id, success && (value.result == :success)])
```

Comment:
In this case we should probably keep it as it is, as the new way wouldn't work once the action gets suspended.

Reply:
Yeah right, we would need to propagate the information about parent action to through the execution plan and ping from there :(